index.js

```diff
@@ -39,73 +39,103 @@ const getPackageJson = function() {
   return pkg;
 };
 
-const defaultGlobals = {
-  browser: {
-    'video.js': 'videojs',
-    'global': 'window',
-    'global/window': 'window',
-    'global/document': 'document'
-  },
-  module: {
-    'video.js': 'videojs'
-  },
-  test: {
-    'qunit': 'QUnit',
-    'qunitjs': 'QUnit',
-    'sinon': 'sinon',
-    'video.js': 'videojs'
-  }
-};
-
-const defaultExternals = {
-  browser: [],
-  module: [
-    'global',
-    'global/document',
-    'global/window'
-  ],
-  test: []
-};
-
-const defaultPlugins = {
-  browser: [
-    'resolve',
-    'json',
-    'commonjs',
-    'uglify',
-    'babel'
-  ],
-
-  module: [
-    'resolve',
-    'json',
-    'commonjs',
-    'babel'
-  ],
+// if checkWatch is true and -w or --watch was passed to rollup
+// we should "change the build" for rollups watch mode.
+const shouldChangeWatch = (settings) =>
+  settings.checkWatch && process.argv.some((arg) => (/^-w|--watch$/).test(arg));
+
+// verify that listed plugins are an object or in primed plugins
+const validatePlugins = ({plugins, primedPlugins}) =>
+  Object.keys(plugins).forEach((buildType) => plugins[buildType].forEach((pluginName) => {
+    if (typeof pluginName !== 'string') {
+      return;
+    }
 
-  test: [
-    'multiEntry',
-    'resolve',
-    'json',
-    'commonjs',
-    'istanbul',
-    'babel'
-  ]
+    if (!primedPlugins[pluginName]) {
+      throw new Error(`Plugin ${pluginName} listed in ${buildType} does not have an equivent Primed Plugin`);
+    }
+  }));
+
+// all complex defaults
+// Note: the order here matters because one default often
+// determines the value of another default
+const ORDERED_DEFAULTS = {
+  globals: () => ({
+    browser: {
+      'video.js': 'videojs',
+      'global': 'window',
+      'global/window': 'window',
+      'global/document': 'document'
+    },
+    module: {
+      'video.js': 'videojs'
+    },
+    test: {
+      'qunit': 'QUnit',
+      'qunitjs': 'QUnit',
+      'sinon': 'sinon',
+      'video.js': 'videojs'
+    }
+  }),
+  externals: ({globals = {}}) => ({
+    browser: [
+    ].concat(Object.keys(globals.browser || {})),
+    module: [
+      'global',
+      'global/document',
+      'global/window'
+    ].concat(Object.keys(globals.module || {})),
+    test: [
+    ].concat(Object.keys(globals.test || {}))
+  }),
+  plugins: (settings) => ({
+    browser: [
+      'resolve',
+      'json',
+      'commonjs',
+      'uglify',
+      'babel'
+    ],
+    module: [
+      'resolve',
+      'json',
+      'commonjs',
+      'babel'
+    ],
+    test: [
+      'multiEntry',
+      'resolve',
+      'json',
+      'commonjs',
+      'istanbul',
+      'babel'
+    ]
+    // filter out istanbul during watch or if coverage is false
+      .filter((n) => n !== 'istanbul' || (!shouldChangeWatch(settings)))
```

Comment:
This is the only real change to the "defaults". We remove istanbul if coverage is false or  shouldChangeWatch is true.